hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupRecordBuffer.java

```diff
@@ -110,7 +112,14 @@ protected FileGroupRecordBuffer(HoodieReaderContext<T> readerContext,
       this.payloadClass = Option.empty();
     }
     this.orderingFieldName = orderingFieldName;
+    // Ensure that ordering field is populated for mergers and legacy payloads
+    orderingFieldName.ifPresent(orderingField -> {
```

Comment:
is the logic consistent with `HoodieFileGroupReader`? should we just update the props in the `HoodieFileGroupReader` itself?

Reply:
The file group reader doesn't handle any of the merging so it makes more sense to keep this all in the `FileGroupRecordBuffer`.  The `orderingFieldName` is just used in the FileGroupReader as an input when constructing the BufferedRecord iterator

Reply:
It is passed to the file group buffer too, at least we make the logic consistent here. I don't think we need a change in `HoodieFileGroupReader`.

Reply:
Ok, I've reverted the changes to the HoodieFileGroupReader